### PR TITLE
Fix assertion failure in semantic model for file-local extension methods

### LINKED ISSUE #67940: IDE errors with file-local extension method
**Version Used**: 

Any.
(I checked with VS 17.5.4 and VS 17.6.0 Preview 4.)

**Steps to Reproduce**:

1. Write followin 3 cs files:

Extension.cs

```cs
internal static class Extensions
{
    public static void M(this string _) { }
}
```

FileExtensions.cs

```cs
file static class FileExtensions
{
    public static void M(this string _) { }
}
```

Program.cs

```cs
"".M();
```

2. Open Program.cs or bring the cursor over `M` in Program.cs

**Expected Behavior**:

No error.

**Actual Behavior**:

image

The IDE repeatedly fails.

The IDE errors can occur even without internal `Extensions`; `Extensions` exists only to ensure successful compilation.

### PR DESCRIPTION
**Pull request by RikkiGibson**
Closes #67940

When lookup gives us a symbol contained in a file-local type, and the lookup is not occurring within the same file, we use CheckViability to essentially filter it out as though the symbol doesn't exist.

The semantic model isn't expecting extension method lookup to yield a symbol, which is then "eliminated" by CheckViability--it's only expecting lookup results which are either valid or that still yield the symbol but also indicate some error condition ("not accessible", etc.). This PR changes that to simply bail out when CheckViability filters out the symbol.

An alternative way to solve this would be to give a non-Empty LookupResultKind for an effectively file-local symbol from another file and an associated diagnostic. Instead of saying "XYZ symbol doesn't exist", the diagnostic would say "XYZ symbol is not in scope in this file" or something like that. Then the original change in this PR wouldn't be necessary as the candidate symbol would still be surfaced and used. That would be a wider-reaching change and could require adding public API, assuming we don't roll it up in the existing "inaccessible" lookup result.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs [hunk 1]
```diff
@@ -4707,8 +4707,11 @@ internal static ImmutableArray<MethodSymbol> GetReducedAndFilteredMethodGroupSym
             ref LookupResultKind resultKind,
             CSharpCompilation compilation)
         {
-            Debug.Assert(singleResult.Kind != LookupResultKind.Empty);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs:4710]: >Debug.Assert(singleResult.Kind != LookupResultKind.Empty); [](http://example.com/codeflow?start=12&length=58)

Is it worth keeping this assert, i.e. moving it below the ```if (singleResult.Symbol is null)``` check?

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs [hunk 2]
```diff
@@ -3693,6 +3693,15 @@ public static void M(this string s)
             // (1,5): error CS1061: 'string' does not contain a definition for 'M' and no accessible extension method 'M' accepting a first argument of type 'string' could be found (are you missing a using directive or an assembly reference?)
             // "a".M(); // 1
             Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "M").WithArguments("string", "M").WithLocation(1, 5));
+
+        var tree = comp.SyntaxTrees[0];
+        var methodNameSyntax = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Single();
+        var model = comp.GetSemanticModel(tree);
+
+        var symbolInfo = model.GetSymbolInfo(methodNameSyntax);
+        Assert.Null(symbolInfo.Symbol);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs:3702]: >Symbol [](http://example.com/codeflow?start=31&length=6)

Consider asserting ```CandidateSyumbols``` and ```CandidateReason```

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

File: src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs:4710]: I thought that if we had a non-null Symbol, then we can be sure we don't have Kind == LookupResultKind.Empty, and there isn't a need to check. I could be wrong about that, though.

**Review by cston — APPROVED** [review]

**Pull request merged**